http/src/main/java/io/micronaut/http/HttpResponse.java

```diff
@@ -36,7 +36,9 @@ public interface HttpResponse<B> extends HttpMessage<B> {
 
     /**
      * @return The current status
+     * @deprecated To support custom status codes. Use {@link #code()} instead of {@link #getStatus()} and {@link HttpStatus#getCode()}  and {@link #reason()} instead of {@link #getStatus()} and {@link HttpStatus#getReason()}
```

Comment:
```suggestion
     * To support custom status codes. Use {@link #code()} instead of {@link #getStatus()} and {@link HttpStatus#getCode()}  and {@link #reason()} instead of {@link #getStatus()} and {@link HttpStatus#getReason()}
```